File: Number_Of_Steps_To_Reduce_A_Number_To_Zero/py.d/steps_reduce_zero.py
class Solution:
  def numberOfSteps (self, num: int) -> int:
    counter = 0
    while num > 0:
      if num % 2 == 0:
        num /= 2
      else:
        num -= 1
      counter += 1
    return int(counter)

if __name__=="__main__":
  solution = Solution()
  print(solution.numberOfSteps(123)) # => 12

File: Number_Of_Steps_To_Reduce_A_Number_To_Zero/js.d/steps_reduce_zero.js

/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps  = function(num) {
  let counter = 0;
  while (num > 1){
    if (num % 2 == 0){
      num /= 2;
      counter += 1;
    } else {
      num -= 1;
      counter += 1;
    }
  }
  counter +=1;
  return counter;
};

console.log(numberOfSteps(123)) // => 12
